tsl/test/expected/data_node.out

```diff
@@ -1813,17 +1830,103 @@ ORDER BY 1, 2;
  hyper_1dim      | _dist_hyper_13_25_chunk | {data_node_2,data_node_3}
 (2 rows)
 
+-- Updates/Deletes should also work
+UPDATE hyper3 SET temp = 10 WHERE time = '2022-01-03 00:00:00';
+UPDATE hyper3 SET temp = 10 WHERE time = '2022-01-03 00:00:05';
+UPDATE hyper_1dim SET temp = 10 WHERE time = '2022-01-03 00:00:00';
+UPDATE hyper_1dim SET temp = 10 WHERE time = '2022-01-03 00:00:05';
+DELETE FROM hyper3 WHERE time = '2022-01-03 00:00:00';
+DELETE FROM hyper3 WHERE time = '2022-01-03 00:00:05';
+DELETE FROM hyper_1dim WHERE time = '2022-01-03 00:00:00';
+DELETE FROM hyper_1dim WHERE time = '2022-01-03 00:00:05';
+-- Inserts directly into chunks using FDW should also work
+INSERT INTO _timescaledb_internal._dist_hyper_12_24_chunk VALUES ('2022-01-11 00:00:00', 1, 1);
+INSERT INTO _timescaledb_internal._dist_hyper_13_25_chunk VALUES ('2022-01-11 00:00:00', 1, 1);
+SELECT hypertable_name, chunk_name, data_nodes FROM timescaledb_information.chunks
+WHERE hypertable_name IN ('hyper3', 'hyper_1dim')
+AND range_start::timestamptz <= '2022-01-10 00:00:00'
+AND range_end::timestamptz > '2022-01-10 00:00:00'
+ORDER BY 1, 2;
+ hypertable_name |       chunk_name        |        data_nodes         
+-----------------+-------------------------+---------------------------
+ hyper3          | _dist_hyper_12_24_chunk | {data_node_2,data_node_3}
+ hyper_1dim      | _dist_hyper_13_25_chunk | {data_node_2,data_node_3}
+(2 rows)
+
+-- DDL should error out even if one DN is unavailable
+\set ON_ERROR_STOP 0
+ALTER TABLE hyper3 ADD COLUMN temp2 int;
+ERROR:  some data nodes are not available for DDL queries
+ALTER TABLE hyper_1dim ADD COLUMN temp2 int;
+ERROR:  some data nodes are not available for DDL queries
+\set ON_ERROR_STOP 1
+-- Mark all DNs unavailable. Metadata should still retain last DN but all
+-- activity should fail
+SELECT * FROM alter_data_node('data_node_2', available=>false);
+WARNING:  could not switch data node on 2 chunks
+  node_name  |   host    | port  |    database    | available 
+-------------+-----------+-------+----------------+-----------
+ data_node_2 | localhost | 55432 | db_data_node_2 | f
+(1 row)
+
+SELECT * FROM alter_data_node('data_node_3', available=>false);
+WARNING:  could not switch data node on 11 chunks
+  node_name  |   host    | port  |    database    | available 
+-------------+-----------+-------+----------------+-----------
+ data_node_3 | localhost | 55432 | db_data_node_3 | f
+(1 row)
+
+\set ON_ERROR_STOP 0
+INSERT INTO hyper3 VALUES ('2022-01-10 00:00:00', 1, 1);
+ERROR:  insufficient number of data nodes
```

Comment:
Should we distinguish between "insufficient number of data nodes" and "insufficient number of available data nodes"? In other words, should we raise the latter when there are nodes that are marked "unavailable" as opposed to them being "deleted"?